ledger/internal/evalprefetcher_test.go

```diff
@@ -0,0 +1,605 @@
+				},
+				{
+					address:        acctAddrPtr(15),
+					creatableIndex: 2001,
+					creatableType:  basics.AppCreatable,
+					resource:       &ledgercore.AccountResource{},
+				},
+				{
+					address:        nil,
+					creatableIndex: 2002,
+					creatableType:  basics.AppCreatable,
+					resource:       nil,
+				},
+				/* - if we'll decide that we want to perfetch the account local state, then this should be enabled.
+				{
+					address:        acctAddrPtr(1),
+					creatableIndex: 10,
+					creatableType:  basics.AppCreatable,
+					resource:       &ledgercore.AccountResource{},
+				},*/
+				{
+					address:        nil,
+					creatableIndex: 10,
+					creatableType:  basics.AppCreatable,
+					resource:       nil,
+				},
+			},
+		},
+	}
+
+	for _, txn := range testTransactions {
+		groups := make([][]transactions.SignedTxnWithAD, 1)
+		groups[0] = make([]transactions.SignedTxnWithAD, 1)
+		groups[0][0].SignedTxn = txn.signedTxn
+
+		preloadedTxnGroupsCh := prefetchAccounts(context.Background(), ledger, rnd, groups, feeSinkAddr, config.Consensus[protocol.ConsensusCurrentVersion])
+
+		for loadedTxnGroup := range preloadedTxnGroupsCh {
+			require.NoError(t, loadedTxnGroup.err)
+
+			// compare the txn.accounts and loadedTxnGroup.accounts in order agnostic way.
+			require.Equal(t, len(txn.accounts), len(loadedTxnGroup.accounts))
+			for _, acct := range txn.accounts {
+				// make sure we find it in loadedTxnGroup.accounts
+				found := false
+				require.NotNil(t, acct.address)
+				for k, loadedAcct := range loadedTxnGroup.accounts {
+					require.NotNilf(t, loadedAcct.address, "index: %d\nexpected %#v\nactual %#v", k, acct, loadedAcct)
+					if *acct.address != *loadedAcct.address {
+						continue
+					}
+					require.Equal(t, *acct.data, *loadedAcct.data)
+					found = true
+					break
+				}
+				require.Truef(t, found, "missing account %#v", acct)
+			}
+
+			// compare the txn.resources and loadedTxnGroup.resources in order agnostic way
+			require.Equalf(t, len(txn.resources), len(loadedTxnGroup.resources), "mismatching resources count; actual : %v", loadedTxnGroup.resources)
+			for _, res := range txn.resources {
+				// make sure we find it in loadedTxnGroup.resources
+				found := false
+				for _, loadedRes := range loadedTxnGroup.resources {
+					if res.creatableIndex != loadedRes.creatableIndex {
+						continue
+					}
+					require.Equal(t, res.creatableType, loadedRes.creatableType)
+					if res.address == nil {
+						require.Nil(t, loadedRes.address)
+					} else {
+						if loadedRes.address == nil || *res.address != *loadedRes.address {
+							continue
+						}
+					}
+					if res.resource == nil {
+						require.Nil(t, loadedRes.resource)
+					} else {
+						require.NotNil(t, loadedRes.resource)
+						require.Equal(t, *res.resource, *loadedRes.resource)
+					}
+					found = true
+					break
+				}
+				require.Truef(t, found, "failed to find resource %#v", res)
+			}
+		}
+	}
+}
```

Comment:
please add one more tests that exercises prefetch queue growth with `expand`ing it couple times, and ensuring that all input random txn groups were piped out the output channel.

Reply:
added.